api/agent/agent.go

```diff
@@ -278,55 +284,95 @@ func (a *agent) getSlot(ctx context.Context, call *call) (Slot, error) {
 	return a.launchCold(ctx, call)
 }
 
-// launchHot checks with slot queue to see if a new container needs to be launched and waits
-// for available slots in the queue for hot request execution.
-func (a *agent) launchHot(ctx context.Context, call *call) (Slot, error) {
+// hotLauncher is spawned in a go routine for each slot queue to monitor stats and launch hot
+// containers if needed. Upon shutdown or activity timeout, hotLauncher exits and during exit,
+// it destroys the slot queue.
+func (a *agent) hotLauncher(ctx context.Context, call *call) {
+
+	// Let use 60 minutes or 2 * IdleTimeout as hot queue idle timeout, pick
```

Comment:
IdleTimeout is quite tricky, it can be zero, so maybe wait for two default idle timeouts (which is 1 minute).
I think that 60 minutes is quite a big timeout, so there should be the ability to configure it from outside, i.e. through Fn opts.

Reply:
60 mins is good. Note that this is the timeout we cache hot container slot queue *and* the relevant stats and we use those stats to decide to launch new containers or not.

Reply:
If we use 1-2 minutes, by the time we end up with useful stats (start/run/wait) for the hot queue, it'll expire.

Reply:
In case of sync mode, would that mean that user would get `Service busy` and will silently die for async?

Reply:
There's a general problem of hardcoded constants in fn. See resource.go, call.go, async.go. I could take all of that out to opts/cfg in a separate PR.

Reply:
It's a good idea, but let's hear other voices on this.

Reply:
If we destroy the queues frequently, this erases those stats. In that case, isNewContainerNeeded() in slots.go becomes a less effective. The cost of making a bad decision is usually a timeout. It would be good to store the stats forever. But also need to avoid bloating the memory with unused hot queues. I think 60 mins is ok. But I agree it should be tune-able somewhere along with other (and more critical) constants.

Reply:
this logic looks fine wrt timing. there's more value in picking sane values for your users than having a bajillion fricking flags to configure every little piece of the server that they don't need to know details of.

Reply:
fwiw idle_timeout will be 30s 99% of the time so this will be 1 minute, as highlighted (perhaps clamp around 60 minutes?)

Reply:
nvm, it is. coffee